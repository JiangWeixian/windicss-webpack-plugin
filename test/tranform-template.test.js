"use strict";

const getModuleSource = require('./helpers/getModuleSource')
const vueWebpackCompiler = require ('./helpers/vueWebpackCompiler')
const reactWebpackCompiler = require ('./helpers/reactWebpackCompiler')

describe("Transform template test", function() {

  it("should render jsx", done => {
    const compiler = reactWebpackCompiler()
    compiler.run((err, stats) => {
      expect(stats.compilation.errors).toStrictEqual([])
      let transformedLayout = getModuleSource('layout.jsx', stats)
      expect(transformedLayout).not.toContain('@apply')
      expect(transformedLayout).toMatchSnapshot('layout1')
      transformedLayout = getModuleSource('layout2.jsx', stats)
      expect(transformedLayout).not.toContain('@apply')
      expect(transformedLayout).toMatchSnapshot('layout2')
      // check the windi file has generated the right classes
      const windi = getModuleSource('virtual:windi.css', stats)
      expect(windi).toContain('bg-blue-500')
      expect(windi).toMatchSnapshot('vue windi')
      done(err)
    });
  });

  it('should render vue', done => {
    const compiler = vueWebpackCompiler()
    compiler.run((err, stats) => {
      expect(stats.compilation.errors).toStrictEqual([])
      const transformedLayout = getModuleSource('App.vue?vue&type=template', stats)
      expect(transformedLayout).not.toContain('hover:(text-green-100 rounded-full bg-teal-900)')
      expect(transformedLayout).toMatchSnapshot('vue template')
      // check the windi file has generated the right classes
      const windi = getModuleSource('virtual:windi.css', stats)
      expect(windi).toContain('text-green-100')
      expect(windi).toMatchSnapshot('vue windi')

      const scss = getModuleSource('main.scss', stats)
      expect(scss).not.toContain('@apply')
      expect(scss).toMatchSnapshot('vue scss')

      const sass = getModuleSource('main.sass', stats)
      expect(sass).not.toContain('@apply')
      expect(sass).toMatchSnapshot('vue sass')

      const less = getModuleSource('main.less', stats)
      expect(less).not.toContain('@apply')
      expect(less).toMatchSnapshot('vue less')

      const css = getModuleSource('plain.css', stats)
      expect(css).not.toContain('@apply')
      expect(css).toMatchSnapshot('vue css')
      done(err)
    });
  });
});
