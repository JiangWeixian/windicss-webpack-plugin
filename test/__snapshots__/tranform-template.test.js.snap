// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transform template test should render jsx: layout1 1`] = `
"import 'virtual:windi.css';
export default function Layout({
  title,
  children
}) {
  return /*#__PURE__*/React.createElement(\\"div\\", {
    id: \\"layout-wrapper\\",
    className: 'bg-gray-100 text-gray-900 dark:bg-gray-900 dark:text-gray-100'
  }, /*#__PURE__*/React.createElement(Head, null, /*#__PURE__*/React.createElement(\\"title\\", null, title)), children, /*#__PURE__*/React.createElement(\\"style\\", {
    jsx: true,
    global: true
  }, \`body:hover {
  --tw-bg-opacity: 1;
  background-color: rgba(59, 130, 246, var(--tw-bg-opacity));
  font-size: 0.75rem;
  line-height: 1rem;
}
body {
  height: 100vh;
  margin: 0px;
  overflow: hidden;
  padding: 0px;
  width: 100vw;
  font-family: '-apple-system', 'BlinkMacSystemFont', 'Segoe UI',
              'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans',
              'Helvetica Neue', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
\`));
}"
`;

exports[`Transform template test should render jsx: layout2 1`] = `
"import 'virtual:windi.css';
export default function Layout({
  title,
  children
}) {
  return /*#__PURE__*/React.createElement(\\"div\\", {
    id: \\"layout-wrapper\\",
    className: \\"bg-gray-100 text-gray-900\\"
  }, /*#__PURE__*/React.createElement(Head, null, /*#__PURE__*/React.createElement(\\"title\\", null, title)), children, /*#__PURE__*/React.createElement(\\"style\\", {
    jsx: true,
    global: true
  }, \`body:hover {
  --tw-bg-opacity: 1;
  background-color: rgba(59, 130, 246, var(--tw-bg-opacity));
  font-size: 0.75rem;
  line-height: 1rem;
}
body {
  height: 100vh;
  margin: 0px;
  overflow: hidden;
  padding: 0px;
  width: 100vw;
  font-family: '-apple-system', 'BlinkMacSystemFont', 'Segoe UI',
              'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans',
              'Helvetica Neue', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
\`));
}"
`;

exports[`Transform template test should render jsx: vue windi 1`] = `
"// Imports
import ___CSS_LOADER_API_IMPORT___ from \\"../../../node_modules/css-loader/dist/runtime/api.js\\";
var ___CSS_LOADER_EXPORT___ = ___CSS_LOADER_API_IMPORT___(function(i){return i[1]});
// Module
___CSS_LOADER_EXPORT___.push([module.id, \\"/* windicss layer base */\\\\n*, ::before, ::after {\\\\n  -webkit-box-sizing: border-box;\\\\n  box-sizing: border-box;\\\\n  border-width: 0;\\\\n  border-style: solid;\\\\n  border-color: #e5e7eb;\\\\n}\\\\n* {\\\\n  --tw-ring-inset: var(--tw-empty,/*!*/ /*!*/);\\\\n  --tw-ring-offset-width: 0px;\\\\n  --tw-ring-offset-color: #fff;\\\\n  --tw-ring-color: rgba(59, 130, 246, 0.5);\\\\n  --tw-ring-offset-shadow: 0 0 #0000;\\\\n  --tw-ring-shadow: 0 0 #0000;\\\\n  --tw-shadow: 0 0 #0000;\\\\n}\\\\n:root {\\\\n  -moz-tab-size: 4;\\\\n  -o-tab-size: 4;\\\\n  tab-size: 4;\\\\n}\\\\n:-moz-focusring {\\\\n  outline: 1px dotted ButtonText;\\\\n}\\\\n:-moz-ui-invalid {\\\\n  box-shadow: none;\\\\n}\\\\n::moz-focus-inner {\\\\n  border-style: none;\\\\n  padding: 0;\\\\n}\\\\n::-webkit-inner-spin-button, ::-webkit-outer-spin-button {\\\\n  height: auto;\\\\n}\\\\n::-webkit-search-decoration {\\\\n  -webkit-appearance: none;\\\\n}\\\\n::-webkit-file-upload-button {\\\\n  -webkit-appearance: button;\\\\n  font: inherit;\\\\n}\\\\n[type='search'] {\\\\n  -webkit-appearance: textfield;\\\\n  outline-offset: -2px;\\\\n}\\\\nabbr[title] {\\\\n  -webkit-text-decoration: underline dotted;\\\\n  text-decoration: underline dotted;\\\\n}\\\\nbody {\\\\n  margin: 0;\\\\n  font-family: inherit;\\\\n  line-height: inherit;\\\\n}\\\\nhtml {\\\\n  -webkit-text-size-adjust: 100%;\\\\n  font-family: ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont,\\\\\\"Segoe UI\\\\\\",Roboto,\\\\\\"Helvetica Neue\\\\\\",Arial,\\\\\\"Noto Sans\\\\\\",sans-serif,\\\\\\"Apple Color Emoji\\\\\\",\\\\\\"Segoe UI Emoji\\\\\\",\\\\\\"Segoe UI Symbol\\\\\\",\\\\\\"Noto Color Emoji\\\\\\";\\\\n  line-height: 1.5;\\\\n}\\\\n/* windicss layer components */\\\\n\\\\n/* windicss layer utilities */\\\\n.bg-gray-100 {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(243, 244, 246, var(--tw-bg-opacity));\\\\n}\\\\n.dark .dark\\\\\\\\:bg-gray-900 {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(17, 24, 39, var(--tw-bg-opacity));\\\\n}\\\\n.hover\\\\\\\\:bg-blue-500:hover {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(59, 130, 246, var(--tw-bg-opacity));\\\\n}\\\\n.h-100vh {\\\\n  height: 100vh;\\\\n}\\\\n.m-0 {\\\\n  margin: 0px;\\\\n}\\\\n.overflow-hidden {\\\\n  overflow: hidden;\\\\n}\\\\n.p-0 {\\\\n  padding: 0px;\\\\n}\\\\n.text-gray-900 {\\\\n  --tw-text-opacity: 1;\\\\n  color: rgba(17, 24, 39, var(--tw-text-opacity));\\\\n}\\\\n.dark .dark\\\\\\\\:text-gray-100 {\\\\n  --tw-text-opacity: 1;\\\\n  color: rgba(243, 244, 246, var(--tw-text-opacity));\\\\n}\\\\n.w-100vw {\\\\n  width: 100vw;\\\\n}\\", \\"\\"]);
// Exports
export default ___CSS_LOADER_EXPORT___;
"
`;

exports[`Transform template test should render vue: vue css 1`] = `
"// Imports
import ___CSS_LOADER_API_IMPORT___ from \\"../../../../node_modules/css-loader/dist/runtime/api.js\\";
var ___CSS_LOADER_EXPORT___ = ___CSS_LOADER_API_IMPORT___(function(i){return i[1]});
// Module
___CSS_LOADER_EXPORT___.push([module.id, \\".global-css {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(251, 191, 36, var(--tw-bg-opacity));\\\\n  padding: 1rem;\\\\n  --tw-text-opacity: 1;\\\\n  color: rgba(255, 255, 255, var(--tw-text-opacity));\\\\n  width: 25%;\\\\n  -webkit-transition-property: background-color, border-color, color, fill, stroke, opacity, -webkit-box-shadow, -webkit-transform, filter, backdrop-filter;\\\\n  -o-transition-property: background-color, border-color, color, fill, stroke, opacity, box-shadow, transform, filter, backdrop-filter;\\\\n  transition-property: background-color, border-color, color, fill, stroke, opacity, box-shadow, -webkit-box-shadow, transform, -webkit-transform, filter, backdrop-filter;\\\\n  -webkit-transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);\\\\n  -o-transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);\\\\n  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);\\\\n  -webkit-transition-duration: 150ms;\\\\n  -o-transition-duration: 150ms;\\\\n  transition-duration: 150ms;\\\\n}\\\\n.global-css:hover {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(120, 53, 15, var(--tw-bg-opacity));\\\\n  --tw-text-opacity: 1;\\\\n  color: rgba(254, 243, 199, var(--tw-text-opacity));\\\\n}\\\\n.global-css h2 {\\\\n  font-weight: 700;\\\\n  font-size: 0.875rem;\\\\n  line-height: 1.25rem;\\\\n}\\", \\"\\"]);
// Exports
export default ___CSS_LOADER_EXPORT___;
"
`;

exports[`Transform template test should render vue: vue less 1`] = `
"// Imports
import ___CSS_LOADER_API_IMPORT___ from \\"../../../../node_modules/css-loader/dist/runtime/api.js\\";
var ___CSS_LOADER_EXPORT___ = ___CSS_LOADER_API_IMPORT___(function(i){return i[1]});
// Module
___CSS_LOADER_EXPORT___.push([module.id, \\".less-global {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(167, 139, 250, var(--tw-bg-opacity));\\\\n  padding: 1rem;\\\\n  --tw-text-opacity: 1;\\\\n  color: rgba(255, 255, 255, var(--tw-text-opacity));\\\\n  width: 25%;\\\\n  -webkit-transition-property: background-color, border-color, color, fill, stroke, opacity, -webkit-box-shadow, -webkit-transform, filter, backdrop-filter;\\\\n  -o-transition-property: background-color, border-color, color, fill, stroke, opacity, box-shadow, transform, filter, backdrop-filter;\\\\n  transition-property: background-color, border-color, color, fill, stroke, opacity, box-shadow, -webkit-box-shadow, transform, -webkit-transform, filter, backdrop-filter;\\\\n  -webkit-transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);\\\\n  -o-transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);\\\\n  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);\\\\n  -webkit-transition-duration: 150ms;\\\\n  -o-transition-duration: 150ms;\\\\n  transition-duration: 150ms;\\\\n}\\\\n.less-global:hover {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(76, 29, 149, var(--tw-bg-opacity));\\\\n  --tw-text-opacity: 1;\\\\n  color: rgba(237, 233, 254, var(--tw-text-opacity));\\\\n}\\\\n.less-global h2 {\\\\n  font-weight: 700;\\\\n  font-size: 0.875rem;\\\\n  line-height: 1.25rem;\\\\n}\\\\n\\", \\"\\"]);
// Exports
export default ___CSS_LOADER_EXPORT___;
"
`;

exports[`Transform template test should render vue: vue sass 1`] = `
"// Imports
import ___CSS_LOADER_API_IMPORT___ from \\"../../../../node_modules/css-loader/dist/runtime/api.js\\";
var ___CSS_LOADER_EXPORT___ = ___CSS_LOADER_API_IMPORT___(function(i){return i[1]});
// Module
___CSS_LOADER_EXPORT___.push([module.id, \\"\\", \\"\\"]);
// Exports
export default ___CSS_LOADER_EXPORT___;
"
`;

exports[`Transform template test should render vue: vue scss 1`] = `
"// Imports
import ___CSS_LOADER_API_IMPORT___ from \\"../../../../node_modules/css-loader/dist/runtime/api.js\\";
var ___CSS_LOADER_EXPORT___ = ___CSS_LOADER_API_IMPORT___(function(i){return i[1]});
// Module
___CSS_LOADER_EXPORT___.push([module.id, \\".scss-global {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(96, 165, 250, var(--tw-bg-opacity));\\\\n  padding: 1rem;\\\\n  --tw-text-opacity: 1;\\\\n  color: rgba(255, 255, 255, var(--tw-text-opacity));\\\\n  width: 25%;\\\\n  -webkit-transition-property: background-color, border-color, color, fill, stroke, opacity, -webkit-box-shadow, -webkit-transform, filter, backdrop-filter;\\\\n  -o-transition-property: background-color, border-color, color, fill, stroke, opacity, box-shadow, transform, filter, backdrop-filter;\\\\n  transition-property: background-color, border-color, color, fill, stroke, opacity, box-shadow, -webkit-box-shadow, transform, -webkit-transform, filter, backdrop-filter;\\\\n  -webkit-transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);\\\\n  -o-transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);\\\\n  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);\\\\n  -webkit-transition-duration: 150ms;\\\\n  -o-transition-duration: 150ms;\\\\n  transition-duration: 150ms;\\\\n}\\\\n\\\\n.scss-global:hover {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(30, 58, 138, var(--tw-bg-opacity));\\\\n  --tw-text-opacity: 1;\\\\n  color: rgba(219, 234, 254, var(--tw-text-opacity));\\\\n}\\\\n\\\\n.scss-global h2 {\\\\n  font-weight: 700;\\\\n  font-size: 0.875rem;\\\\n  line-height: 1.25rem;\\\\n}\\", \\"\\"]);
// Exports
export default ___CSS_LOADER_EXPORT___;
"
`;

exports[`Transform template test should render vue: vue template 1`] = `
"var render = function() {
  var _vm = this
  var _h = _vm.$createElement
  var _c = _vm._self._c || _h
  return _c(\\"div\\", { attrs: { id: \\"app\\" } }, [
    _c(\\"nav\\", [_vm._v(\\"\\\\n\\\\t\\\\t\\\\t\\\\tWindi test CSS\\\\n\\\\t\\\\t\\\\t\\")]),
    _vm._v(\\" \\"),
    _c(
      \\"div\\",
      {
        staticClass:
          \\"px-5 p-1 text-red hover:text-green-100 hover:rounded-full hover:bg-teal-900\\",
        class: { \\"bg-teal-100\\": true, \\"hover:w-1/3\\": true }
      },
      [_vm._v(\\"\\\\n\\\\t\\\\t\\\\t\\\\tHello World\\\\n\\\\t\\\\t\\\\t\\")]
    ),
    _vm._v(\\" \\"),
    _c(\\"h2\\", { staticClass: \\"btn\\" }, [_vm._v(\\"\\\\n\\\\t\\\\t\\\\t\\\\tButton\\\\n\\\\t\\\\t\\\\t\\")]),
    _vm._v(\\" \\"),
    _c(\\"div\\", { staticClass: \\"bar\\" }, [_vm._v(\\"\\\\n\\\\t\\\\t\\\\t\\\\tBar\\\\n\\\\t\\\\t\\\\t\\")]),
    _vm._v(\\" \\"),
    _c(\\"div\\", { staticClass: \\"bg-black bg-blue-200 bg-red-200\\" }, [
      _vm._v(\\"\\\\n\\\\t\\\\t\\\\t\\\\tWithout ! (red)\\\\n\\\\t\\\\t\\\\t\\")
    ]),
    _vm._v(\\" \\"),
    _c(\\"div\\", { staticClass: \\"bg-black !bg-blue-200 bg-red-200\\" }, [
      _vm._v(\\"\\\\n\\\\t\\\\t\\\\t\\\\tWith ! (blue)\\\\n\\\\t\\\\t\\\\t\\")
    ]),
    _vm._v(\\" \\"),
    _vm._m(0),
    _vm._v(\\" \\"),
    _vm._m(1),
    _vm._v(\\" \\"),
    _vm._m(2),
    _vm._v(\\" \\"),
    _vm._m(3),
    _vm._v(\\" \\"),
    _c(
      \\"button\\",
      {
        attrs: {
          bg: \\"blue-400 hover:blue-500 dark:blue-500 dark:hover:blue-600\\",
          text: \\"sm white\\",
          font: \\"mono light\\",
          p: \\"y-2 x-4\\",
          border: \\"2 rounded blue-200\\"
        }
      },
      [_vm._v(\\"\\\\n      Button\\\\n    \\")]
    )
  ])
}
var staticRenderFns = [
  function() {
    var _vm = this
    var _h = _vm.$createElement
    var _c = _vm._self._c || _h
    return _c(\\"div\\", { staticClass: \\"less-global\\" }, [
      _c(\\"h2\\", [_vm._v(\\"Less global\\")])
    ])
  },
  function() {
    var _vm = this
    var _h = _vm.$createElement
    var _c = _vm._self._c || _h
    return _c(\\"div\\", { staticClass: \\"scss-global\\" }, [
      _c(\\"h2\\", [_vm._v(\\"SCSS global\\")])
    ])
  },
  function() {
    var _vm = this
    var _h = _vm.$createElement
    var _c = _vm._self._c || _h
    return _c(\\"div\\", { staticClass: \\"sass-global\\" }, [
      _c(\\"h2\\", [_vm._v(\\"SASS global\\")])
    ])
  },
  function() {
    var _vm = this
    var _h = _vm.$createElement
    var _c = _vm._self._c || _h
    return _c(\\"div\\", { staticClass: \\"css-global\\" }, [
      _c(\\"h2\\", [_vm._v(\\"CSS global\\")])
    ])
  }
]
render._withStripped = true

export { render, staticRenderFns }"
`;

exports[`Transform template test should render vue: vue windi 1`] = `
"// Imports
import ___CSS_LOADER_API_IMPORT___ from \\"../../../node_modules/css-loader/dist/runtime/api.js\\";
var ___CSS_LOADER_EXPORT___ = ___CSS_LOADER_API_IMPORT___(function(i){return i[1]});
// Module
___CSS_LOADER_EXPORT___.push([module.id, \\"/* windicss layer base */\\\\n*, ::before, ::after {\\\\n  -webkit-box-sizing: border-box;\\\\n  box-sizing: border-box;\\\\n  border-width: 0;\\\\n  border-style: solid;\\\\n  border-color: #e5e7eb;\\\\n}\\\\n* {\\\\n  --tw-ring-inset: var(--tw-empty,/*!*/ /*!*/);\\\\n  --tw-ring-offset-width: 0px;\\\\n  --tw-ring-offset-color: #fff;\\\\n  --tw-ring-color: rgba(59, 130, 246, 0.5);\\\\n  --tw-ring-offset-shadow: 0 0 #0000;\\\\n  --tw-ring-shadow: 0 0 #0000;\\\\n  --tw-shadow: 0 0 #0000;\\\\n}\\\\n:root {\\\\n  -moz-tab-size: 4;\\\\n  -o-tab-size: 4;\\\\n  tab-size: 4;\\\\n}\\\\n:-moz-focusring {\\\\n  outline: 1px dotted ButtonText;\\\\n}\\\\n:-moz-ui-invalid {\\\\n  box-shadow: none;\\\\n}\\\\n::moz-focus-inner {\\\\n  border-style: none;\\\\n  padding: 0;\\\\n}\\\\n::-webkit-inner-spin-button, ::-webkit-outer-spin-button {\\\\n  height: auto;\\\\n}\\\\n::-webkit-search-decoration {\\\\n  -webkit-appearance: none;\\\\n}\\\\n::-webkit-file-upload-button {\\\\n  -webkit-appearance: button;\\\\n  font: inherit;\\\\n}\\\\n[type='search'] {\\\\n  -webkit-appearance: textfield;\\\\n  outline-offset: -2px;\\\\n}\\\\nabbr[title] {\\\\n  -webkit-text-decoration: underline dotted;\\\\n  text-decoration: underline dotted;\\\\n}\\\\nbody {\\\\n  margin: 0;\\\\n  font-family: inherit;\\\\n  line-height: inherit;\\\\n}\\\\nbutton {\\\\n  font-family: inherit;\\\\n  font-size: 100%;\\\\n  line-height: 1.15;\\\\n  margin: 0;\\\\n  text-transform: none;\\\\n  background-color: transparent;\\\\n  background-image: none;\\\\n  padding: 0;\\\\n  line-height: inherit;\\\\n  color: inherit;\\\\n}\\\\nbutton, [type='button'], [type='reset'], [type='submit'] {\\\\n  -webkit-appearance: button;\\\\n}\\\\nbutton:focus {\\\\n  outline: 1px dotted;\\\\n  outline: 5px auto -webkit-focus-ring-color;\\\\n}\\\\nbutton, [role=\\\\\\"button\\\\\\"] {\\\\n  cursor: pointer;\\\\n}\\\\nhtml {\\\\n  -webkit-text-size-adjust: 100%;\\\\n  font-family: ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont,\\\\\\"Segoe UI\\\\\\",Roboto,\\\\\\"Helvetica Neue\\\\\\",Arial,\\\\\\"Noto Sans\\\\\\",sans-serif,\\\\\\"Apple Color Emoji\\\\\\",\\\\\\"Segoe UI Emoji\\\\\\",\\\\\\"Segoe UI Symbol\\\\\\",\\\\\\"Noto Color Emoji\\\\\\";\\\\n  line-height: 1.5;\\\\n}\\\\nh2 {\\\\n  margin: 0;\\\\n  font-size: inherit;\\\\n  font-weight: inherit;\\\\n}\\\\n/* windicss layer components */\\\\n\\\\n/* windicss layer utilities */\\\\n.hover\\\\\\\\:bg-teal-900:hover {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(19, 78, 74, var(--tw-bg-opacity));\\\\n}\\\\n.bg-teal-100 {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(204, 251, 241, var(--tw-bg-opacity));\\\\n}\\\\n.bg-black {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(0, 0, 0, var(--tw-bg-opacity));\\\\n}\\\\n.bg-blue-200 {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(191, 219, 254, var(--tw-bg-opacity));\\\\n}\\\\n.bg-red-200 {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(254, 202, 202, var(--tw-bg-opacity));\\\\n}\\\\n.\\\\\\\\!bg-blue-200 {\\\\n  --tw-bg-opacity: 1 !important;\\\\n  background-color: rgba(191, 219, 254, var(--tw-bg-opacity)) !important;\\\\n}\\\\n.hover\\\\\\\\:rounded-full:hover {\\\\n  border-radius: 9999px;\\\\n}\\\\n.rounded {\\\\n  border-radius: 0.25rem;\\\\n}\\\\n.p-1 {\\\\n  padding: 0.25rem;\\\\n}\\\\n.px-5 {\\\\n  padding-left: 1.25rem;\\\\n  padding-right: 1.25rem;\\\\n}\\\\n.hover\\\\\\\\:text-green-100:hover {\\\\n  --tw-text-opacity: 1;\\\\n  color: rgba(209, 250, 229, var(--tw-text-opacity));\\\\n}\\\\n.hover\\\\\\\\:w-1\\\\\\\\/3:hover {\\\\n  width: 33.333333%;\\\\n}\\", \\"\\"]);
// Exports
export default ___CSS_LOADER_EXPORT___;
"
`;
